pkg/prometheus/promcfg.go

```diff
@@ -2289,29 +2293,38 @@ func (cg *ConfigGenerator) appendEvaluationInterval(slice yaml.MapSlice, evaluat
 	return append(slice, yaml.MapItem{Key: "evaluation_interval", Value: evaluationInterval})
 }
 
+func (cg *ConfigGenerator) appendGlobalLimits(slice yaml.MapSlice, limitKey string, limit *uint64, enforcedLimit *uint64) yaml.MapSlice {
+	if ptr.Deref(limit, 0) > 0 {
+		if ptr.Deref(enforcedLimit, 0) > 0 && *limit < *enforcedLimit {
```

Comment:
shouldn't it be `*limit > *enforcedLimit`?

Reply:
Don't think so!
Sample Limit == Soft Limit

Reply:
I'd rather expect the opposite: enforced (hard) limits should always be greater than global (soft) limits?
In any case it should be documented in the `sampleLimits` field.